components/api_version/src/api_v1.rs

```diff
@@ -35,4 +35,35 @@ impl KvFormat for ApiV1 {
     fn encode_raw_value_owned(value: RawValue<Vec<u8>>) -> Vec<u8> {
         value.user_value
     }
+
+    fn convert_encoded_key_version_from(
+        src_api: ApiVersion,
+        key: &[u8],
+        _ts: Option<TimeStamp>,
+    ) -> Result<Key> {
+        match src_api {
+            ApiVersion::V1 | ApiVersion::V1ttl => Ok(Key::from_encoded_slice(key)),
+            ApiVersion::V2 => {
+                assert_eq!(ApiV2::parse_key_mode(key), KeyMode::Raw);
+                let (mut user_key, _) = ApiV2::decode_raw_key(&Key::from_encoded_slice(key), true)?;
+                user_key.remove(0); // remove first byte `RAW_KEY_PREFIX`
+                Ok(Self::encode_raw_key_owned(user_key, None))
+            }
+        }
+    }
+
+    fn convert_user_key_range_version_from(
+        src_api: ApiVersion,
+        mut start_key: Vec<u8>,
+        mut end_key: Vec<u8>,
+    ) -> (Vec<u8>, Vec<u8>) {
+        match src_api {
+            ApiVersion::V1 | ApiVersion::V1ttl => (start_key, end_key),
+            ApiVersion::V2 => {
+                start_key.remove(0);
```

Comment:
Should we check if the range is valid?

Reply:
the user key should be guranteed by client, as it may be an end of key range, 'r' + 1. So here we cannot check the validition. `encode_raw_key/decode_raw_key` also handle it without check the range validition.

Reply:
Without check, will it corrupt data silently?

Reply:
the user of api_version is inside tikv-server, like BR / CDC / Storage module, They will do the range validation check. The reason that api_version does not check range is that it can accept key which exceed the range like 'r' + 1 as the range end in raw_scan or some other functions.  api_version do check of encoding in `is_valid_encoded_key`.

Reply:
https://github.com/tikv/tikv/blob/222c5761461d4343b52a272c41364502ba9b7729/src/storage/mod.rs#L2020. This shows raw_scan will check the validation of the range. I think ApiV2's range is well defined and can be checked here.

Reply:
OK, we can refact the check_api_version_range to check here. I would like to leave a TODO to do this. Thanks for the good suggestions.